### PR TITLE
Added LetterTileRack class

### PR DESCRIPTION
**Pull request by Peter-Scheelke**
I added the LetterTileRack class, added unit tests for the LetterTile class, the GameBoardSquare class, and the LetterTileRack classes. All of these classes pass Stylecop. I also slightly edited the Player class.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/sh4ne/Scrabble_CS3450/pull/38" target="_blank">ReviewNinja

**Comment by johnsonjo4531** [comment]
I would plus 1 this, but it says "This branch has conflicts that must be resolved". I would guess the best thing to do is for you to resolve them first. I just recently fixed some merge conflicts at my work so I will have a better chance of being able to help if you need it this time.

**Comment by johnsonjo4531** [comment]
#15

File: ScrabbleUnitTest/Game Objects/LetterTileRackTests.cs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+            {
+                exceptionWasThrown = true;
+            }
+
+            Assert.IsTrue(exceptionWasThrown);
+            exceptionWasThrown = false;
+
+        }
+    }
+}
```
Comment (sh4ne) [inline ScrabbleUnitTest/Game Objects/LetterTileRackTests.cs]: _Optional_: It shows that a new line is needed at the end of the file

File: ScrabbleUnitTest/Game Objects/LetterTileTests.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            Assert.IsTrue(lt1.Equals(lt2));
+
+            lt2 = new LetterTile('E', 1);
+            Assert.IsFalse(lt1.Equals(lt2));
+
+            lt2 = new LetterTile('A', 2);
+            Assert.IsFalse(lt1.Equals(lt2));
+        }
+    }
+}
```
Comment (sh4ne) [inline ScrabbleUnitTest/Game Objects/LetterTileTests.cs]: _Optional_: New line here as well, I don't know why this matters but it shows it in the code.

**Comment by sh4ne** [comment]
:+1: **good job!** I'm +1-ing now because merge conflicts will have to be fixed before we can actually merge, so it is already getting checked.

**Comment by Peter-Scheelke** [comment]
Is there an easy way to see what's causing the merge conflicts? I thought it was changes that I made to the Player class, so I reverted those, but it didn't help.

**Comment by sh4ne** [comment]
There is a way to see where the merge conflicts live. Are you using source tree?

**Comment by sh4ne** [comment]
:+1: for changes

**Comment by Peter-Scheelke** [comment]
I am not.

Urgh. I've changed every line that deleted/changed from the files that were already in the project, and it didn't fix anything . . .

**Comment by sh4ne** [comment]
well, I fixed the conflict, and tried to get things pushed to here.... but I pushed them straight to the master, which works.

File: Scrabble/Scrabble.csproj [hunk 3]
```diff
@@ -53,6 +53,7 @@
       <DependentUpon>Form1.cs</DependentUpon>
     </Compile>
     <Compile Include="Game Objects\GameBoardSquare.cs" />
+    <Compile Include="Game Objects\LetterTileRack.cs" />
```
Comment (sh4ne) [inline Scrabble/Scrabble.csproj:56]: This is where the merge conflict was.

**Pull request merged**